Control/Concurrent/STM/TBQueue.hs

```diff
@@ -127,11 +127,11 @@ readTBQueue (TBQueue rsize read _wsize write _size) = do
       return x
     [] -> do
       ys <- readTVar write
-      case ys of
-        [] -> retry
-        _  -> do
-          let (z:zs) = reverse ys -- NB. lazy: we want the transaction to be
-                                  -- short, otherwise it will conflict
+      -- NB. lazy: we want the transaction to be short,
+      -- otherwise it will conflict
+      case reverse ys of
+        []     -> retry
+        (z:zs) -> do
```

Comment:
btw, the `()`s are redundant here

I wonder if the required laziness has been retained